### PR TITLE
Introducing see all promises of political parties

### PR DESCRIPTION
**Pull request by AriHrannar**
Adding promises to political parties

Related to: [https://github.com/veritus/veritus-backend/pull/40](https://github.com/veritus/veritus-backend/pull/40)

### CODE REVIEW

**Comment by AriHrannar** [comment]
@Ragnar-Hardarson need some thoughts from you on how to structure this.

Right now I have a container for PartyPromises that I feed the Party Id into as a props. I then use that prop to call the webservice to get promises made by that party. 

However, right now its complaining that this.props.partyId is not in PropTypes, but if I add that then it complains that PropTypes is depricated :) Any examples of using the Prop thing (from flow?) is in presentational components who are functions, but that doesnt really fit as I have a lifecycle method (componentWillMount) to get the promises

So I am in a bit of a pickle what the right thing to do here is, any suggestions?

**Comment by AriHrannar** [comment]
An idea might be to do this on a details page (party/1) and then use params to get the id that way. Its probably a better long term solution to have a details page as the current list view will get cluttered fast I think.

**Comment by Ragnar-Hardarson** [comment]
I like the structure you've got right here. We can move it around at a later stage if needed.

For Flow type checking I'd recommend https://flow.org/en/docs/frameworks/react/#typing-react-with-flow-a-classtoc-idtoc-typing-react-with-flow-hreftoc-typing-react-with-flowa

I'll have a go at typechecking this

**Comment by Ragnar-Hardarson** [comment]
I should maybe help you setup Flow in your editor.
If you're using VSCode you can use [this](https://marketplace.visualstudio.com/items?itemName=flowtype.flow-for-vscode)

**Comment by Ragnar-Hardarson** [comment]
I did a forced update on this branch to keep it in check with Master.

Please follow these steps
```
git checkout master
git branch -D feature/see-promises-on-political-party
git fetch origin
git checkout feature/see-promises-on-political-party
```

File: src/Containers/PartyPromises/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/* @flow */
+import React from 'react';
+import type { Match } from 'react-router-dom';
+import PromiseList from '../../Components/PromiseList';
+import { getPromisesByPoliticalParty } from '../../utils/api';
+import type { PromiseType } from '../../types';
+
+export type Props = {
+  match: Match,
```
Comment (Ragnar-Hardarson) [inline src/Containers/PartyPromises/index.js]: this could be replaced with
`partyId: PartyIdType`

File: src/Containers/PartyPromises/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+export class PartyPromises extends React.Component {
+  state: {
+    promises: Array<PromiseType>,
+  };
+
+  props: Props;
```
Comment (Ragnar-Hardarson) [inline src/Containers/PartyPromises/index.js]: This line tells React to use the Flow PropTypes

**Comment by Ragnar-H** [comment]
@AriHrannar lets merge #41 to master and update this branch before we merge this.

**Comment by Ragnar-H** [comment]
@arihrannar could you take care of the merge conflicts?

**Comment by Ragnar-H** [comment]
@AriHrannar I got excited and fixed the merge conflicts.
You can have a look at https://github.com/veritus/veritus-web/pull/31/commits/99a867426d7680f74f36b73e59e841641175efbc if you want to see the changes

**Pull request merged**